internal/integration/api/selinux.go

```diff
@@ -336,7 +337,121 @@ func (suite *SELinuxSuite) TestSecurityState() {
 // TODO: test for system and CRI container labels
 // TODO: test labels for unconfined system extensions, pods
 // TODO: test for no avc denials in dmesg
-// TODO: start a pod and ensure access to restricted resources is denied
+
+// TestNoPtrace confirms ptracing system processes is prohibited in enforcing mode.
+func (suite *SELinuxSuite) TestNoPtrace() {
+	if !suite.SelinuxEnforcing {
+		suite.T().Skip("skipping SELinux negative tests in permissive mode")
+	}
+
+	ptraceTestPodDef, err := suite.NewPrivilegedPod("pid1-ptrace-test")
+	suite.Require().NoError(err)
+	ptraceTestPodDef.SetQuiet(true)
+
+	suite.Require().NoError(ptraceTestPodDef.Create(suite.ctx, 5*time.Minute))
+
+	defer ptraceTestPodDef.Delete(suite.ctx) //nolint:errcheck
+
+	_, stderr, err := ptraceTestPodDef.Exec(
+		suite.ctx,
+		"apk add --update strace",
+	)
+
+	suite.Assert().NoError(err)
+	suite.Assert().Empty(stderr, "stderr: %s", stderr)
+
+	// if attached, timeout
+	ctx, cancel := context.WithTimeout(suite.ctx, time.Second*5)
+	defer cancel()
+
+	_, stderr, err = ptraceTestPodDef.Exec(
+		ctx,
+		"strace -p 1",
+	)
+
+	// in case of successful attach it will be context.DeadlineExceeded
+	suite.Assert().NotNil(err)
```

Comment:
```suggestion
	suite.Require().Error(err)
```

(this would abort execution in case error is `nil`, which might panic below)